src/main/java/org/opensearch/security/privileges/SecurityIndexAccessEvaluator.java

```diff
@@ -150,6 +156,23 @@ public PrivilegesEvaluatorResponse evaluate(final ActionRequest request, final T
         return presponse;
     }
 
+    private  boolean checkSystemIndexPermissionsForUser(ConfigModelV7.SecurityRoles securityRoles) {
+        Set<WildcardMatcher> userPermMatchers = new HashSet<>();
+
+        securityRoles.getRoles().stream().forEach(securityRole -> {
+            securityRole.getIpatterns().stream().forEach((ip) -> {
+                userPermMatchers.add(ip.getNonWildCardPerms());
+            });
+        });
+
+        for(WildcardMatcher userPermMatcher : userPermMatchers.stream().filter( matcher ->  !matcher.toString().equals("*")).collect(Collectors.toSet()) ){
```

Comment:
Looks like you probably do not need `!matcher.toString().equals(*)` here since the `userPermMatchers` is constructed from `userPermMatchers.add(ip.getNonWildCardPerms)`. This sounds like you would never have any matchers equal to *.